eng/source-build-patches/0002-Conditionally-set-PackAsToolShimRID.patch

```diff
@@ -0,0 +1,28 @@
+From 64d2efe24308d9e8b99fdea60c92ee1115259506 Mon Sep 17 00:00:00 2001
+From: dseefeld <dseefeld@microsoft.com>
+Date: Thu, 31 Oct 2019 20:38:26 +0000
+Subject: [PATCH 02/14] Conditionally set PackAsToolShimRID
+
+---
+ Directory.Build.targets | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/Directory.Build.targets b/Directory.Build.targets
+index 4b72a52947..1a3b01e858 100644
+--- a/Directory.Build.targets
++++ b/Directory.Build.targets
+@@ -52,6 +52,11 @@
+     <SuppressDependenciesWhenPacking Condition="'$(SuppressDependenciesWhenPacking)' == '' AND '$(IsAnalyzersProject)' == 'true'">true</SuppressDependenciesWhenPacking>
+   </PropertyGroup>
+ 
++  <PropertyGroup Condition="'$(PackAsTool)' == 'true' AND '$(IsShippingPackage)' == 'true' AND '$(DotNetBuildFromSource)' != 'true'">
++    <!-- This is a requirement for Microsoft tool packages only. -->
++    <PackAsToolShimRuntimeIdentifiers>win-x64;win-x86</PackAsToolShimRuntimeIdentifiers>
++  </PropertyGroup>
++
```

Comment:
Seems strange because this patch duplicates existing code in src/Tools/Directory.Build.targets. Should ensure (if we don't already) the tools don't build in the `$(DotNetBuildFromSource)` case. Bottom line, I'd rather we didn't have or need this patch.

Reply:
The patches being introduced here are the current patches that exist in the current source-build.  Our plan as described in the [Arcade Powered Source-Build Implementation Plan](https://github.com/dotnet/source-build/blob/main/Documentation/planning/arcade-powered-source-build/implementation-plan.md#stage-descriptions), is for the repo teams to incorporate/eliminate/refactor these patches after getting the initial source-build infra in place.  Some of these patches may be out of date, some may be better achieved via a different means, etc.  The real key here is that we will have CI legs in each repo to ensure that the repo is always source-buildable.